File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(num)
    out = ''
    out += 'Pling' if num % 3 == 0
    out += 'Plang' if num % 5 == 0
    out += 'Plong' if num % 7 == 0
    return out unless out == ''
    num.to_s
  end
end

module BookKeeping
  VERSION = 2
end

File: javascript/raindrops/raindrops.js
module.exports = function() {
  this.convert = function(num) {
    var drops = [];
    if (num % 3 == 0) drops.push('Pling');
    if (num % 5 == 0) drops.push('Plang');
    if (num % 7 == 0) drops.push('Plong');
    if (drops.length > 0) return drops.join('');
    return num.toString();
  }
}
